controlpanel/frontend/static/javascripts/modules/role-names.js

```diff
@@ -0,0 +1,53 @@
+moj.Modules.roleNames = {
+  id: "role_name",
+  selectId: "role_name-select",
+  formClass: "appRoles",
+  roleListEndpoint: "/parameters/form/role-list.js",
+  roles: null,
+
+  init() {
+    if (document.querySelectorAll("." + this.formClass).length) {
+      this.getRoles().done(() => {
+        this.loadRolesToSelect();
+        this.bindEvents();
+        console.log(this.roles);
+      });
+    }
+    this.$repoSlugTypeahead = $('#' + this.id)
```

Comment:
Avoid using JQuery unnecessarily. This could be `this.repoSlugTypeahead = document.getElementById(this.id);`